go/src/koding/kites/kloud/metadata/provision.sh

```diff
@@ -5,38 +5,72 @@
 # Copyright (C) 2012-2017 Koding Inc., all rights reserved.
 
 set -euo pipefail
+
+# The following variables are passed via terraform's variable block.
+export KODING_USERNAME="$${KODING_USERNAME:-${var.koding_account_profile_nickname}}"
+export KLIENT_URL="$${KLIENT_URL:-${var.koding_klient_url}}"
+export SCREEN_URL="$${SCREEN_URL:-${var.koding_screen_url}}"
+export USE_EMBEDDED="$${USE_EMBEDDED:-${var.koding_use_embedded}}"
+
 trap "echo _KD_DONE_ | tee -a /var/log/cloud-init-output.log" EXIT
 
-echo "127.0.0.1 {{.Hostname}}" >> /etc/hosts
+install_screen() {
+	curl --location --silent --show-error --retry 5 "$${SCREEN_URL}" --output /tmp/screen.tar.gz
+    tar -C / -xf /tmp/screen.tar.gz
+
+	groupadd --force screen
+	usermod -a -G screen "$${KODING_USERNAME}"
+
+	if [ ! -x /usr/bin/screen ]; then
+		ln -sf /opt/kite/klient/embedded/bin/screen /usr/bin/screen
+	fi
+
+	if [ ! -e /usr/share/terminfo ]; then
+		mkdir -p /usr/share
+		ln -sf /opt/kite/klient/embedded/share/terminfo /usr/share/terminfo
+		ln -sf /usr/share/terminfo/X /usr/share/terminfo/x
+	fi
 
-mkdir -p /opt/kite/klient
+	if [ ! -e /var/run/screen ]; then
+		mkdir -p /var/run/screen
+		chmod 0755 /var/run/screen
+		chown root:screen /var/run/screen
+	fi
+}
 
-wget "{{.KlientURL}}" --retry-connrefused --tries 5 -O /tmp/klient.gz
-wget "{{.ScreenURL}}" --retry-connrefused --tries 5 -O /tmp/screen.tar.gz
+main() {
+	echo "127.0.0.1 $${KODING_USERNAME}" >> /etc/hosts
 
-tar -C / -xf /tmp/screen.tar.gz
+	touch /var/log/klient.log \
+	      /var/log/cloud-init-output.log
 
-if [ ! -x /usr/bin/screen ]; then
-	ln -sf /opt/kite/klient/embedded/bin/screen /usr/bin/screen
-fi
+	if ! type screen &>/dev/null; then
+		if [ $${USE_EMBEDDED} -eq 1 ]; then
+			install_screen
+		elif type yum &>/dev/null; then
+			yum install --assumeyes screen
```

Comment:
could we install a specific and tested version of screen otherwise we might have problems and it would be hard to debug since we wont have any idea on which version is installed. Even on AWS some of the images(same version) has different versions of tools.